libcnb-data/src/launch.rs

```diff
@@ -65,30 +66,102 @@ pub struct Label {
 pub struct Process {
     pub r#type: ProcessType,
     pub command: String,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub args: Option<Vec<String>>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub direct: Option<bool>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub default: Option<bool>,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub args: Vec<String>,
+    #[serde(default, skip_serializing_if = "std::ops::Not::not")]
+    pub direct: bool,
+    #[serde(default, skip_serializing_if = "std::ops::Not::not")]
+    pub default: bool,
```

Comment:
So from reading #243 it sounds like the upstream CNB spec does treat proceses with unset values differently from processes with values set to their default values.

That is, it seems we really do need `Option` in order to represent this?

Reply:
Although looking at the lifecycle implementation, it seems that they don't treat the unset case any differently?
https://github.com/buildpacks/lifecycle/blob/6f56a153034317a88f41109b76e026b93134eb89/launch/launch.go#L9-L16

(According to https://pkg.go.dev/encoding/json#Marshal `omitempty` will omit `false` or empty array values)

Reply:
> You're referring to #243 (comment), correct?
Yeah

Reply:
I don't think lifecycle treats it differently b/c this is how Go works.

Reply:
I see. This would be an issue for all other types in `libcnb-data` as well where we use non-`Option` defaults. This never came up before, but if we'd change `libcnb-data` to support that use-case, we should do it for all types for consistency.

I'm hesitant to make this change lightly, as it's a severely breaking change and in many cases complicates the usage  for the more common use-cases since a user would need to resolve the "missing value" case to the default value by hand.

Maybe we can have types suffixed with `Raw` in `libcnb-data` that use an enum for each optional TOML field that is almost identical to `Option` but conveys additional semantics:

```rust
enum<A> OptionalValue<A> {
     Missing,
     Value(A)
}
```

The types without the `Raw` suffix could stay the same and would resolve `OptionalValue`s during deserialization by replacing them with the default values like they do today.

Reply:
Also, whilst the CNB spec does state that there should be a difference between unset and "set to what happens to be the default value", it seems the implementation in `lifecycle` doesn't honour this? See:
https://github.com/Malax/libcnb.rs/pull/265#discussion_r784123333

Reply:
@edmorley This isn't really about the lifecycle. From my understanding, @lead4good wants to parse to `launch.toml` files, and merge them together. Without knowing which field has the a default value, default values might override explicit values.